### PR TITLE
Disable parameter mapping with native models

### LINKED ISSUE #44288: Dashboard filters do not work with SQL models
### Describe the bug

When a dashboard filter is connected to a SQL model, adding a filter doesn't actually filter data.

### To Reproduce

1. New -> Model -> SQL -> `SELECT * FROM PRODUCTS` - run and save
2. Add to a dashboard
3. Add a text filter and connect to the `CATEGORY` field
4. Set the filter value to `Widget` 
5. See that after the filter is applied the query results are the same and obviously not filtered
6. The same seems to happen with any other filter type or column

### Expected behavior

It should actually filter data.

### Logs

No relevant FE or BE logs. It seems that the BE silently ignores some error.

### Information about your Metabase installation

```JSON
v50
```

### Severity

P1/P2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ranquild**
closes https://github.com/metabase/metabase/issues/44288
Demo https://www.loom.com/share/2718b1e2389046858343fa88a12a3b7c

Changes:
- Disallows parameter mapping to native models

How to verify:
- add a SQL (`SELECT * FROM ORDERS`) model to a dashboard and try to map it to a numeric parameter
- you should see a new error message

Screenshot 2024-06-18 at 15 06 41

### CODE REVIEW

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DisabledNativeCardHelpText/tests/common.unit.spec.ts [hunk 1]
```diff
@@ -1,27 +1,65 @@
...
+
+  it.each([{ showMetabaseLinks: false }, { showMetabaseLinks: true }])(
+    "should show a model help link when `show-metabase-links`: %s",
+    ({ showMetabaseLinks }) => {
+      setup({ cardType: "model", showMetabaseLinks });
+      expect(
+        screen.getByRole("link", { name: "Learn more" }),
+      ).toBeInTheDocument();
+    },
+  );
```
Comment (kamilmielnik) [inline frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DisabledNativeCardHelpText/tests/common.unit.spec.ts:64]: I'm confused by assertions in these 2 tests and in `DisabledNativeCardHelpText/tests/enterprise.unit.spec.ts`.
I thought that when `showMetabaseLinks` is `false` we don't want to show these links :thinking: 

Can you please explain?

**Review by kamilmielnik — COMMENTED** [review]
Left a question inline

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DisabledNativeCardHelpText/tests/common.unit.spec.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DisabledNativeCardHelpText/tests/common.unit.spec.ts:64]: This is the existing test. It means that in OSS and EE without a token this setting is ignored.

Comment (kamilmielnik) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DisabledNativeCardHelpText/tests/common.unit.spec.ts:64]: I think it'd be good to leave a comment with this information (or tweak name of the test) :pray:

**Review by kamilmielnik — APPROVED** [review]

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DisabledNativeCardHelpText/tests/common.unit.spec.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DisabledNativeCardHelpText/tests/common.unit.spec.ts]: Updated test names

**Pull request merged**